ethcore/src/block.rs

```diff
@@ -107,13 +108,17 @@ pub struct ExecutedBlock {
 	pub state: State<StateDB>,
 	/// Transaction traces.
 	pub traces: Tracing,
+	/// Writer for persisting storage changes.
+	pub storage_writer: Box<storage_writer::StorageWriter>,
+	/// Block hash.
+	pub header_hash: H256,
```

Comment:
It's already part of `self.header.hash`

Reply:
The hash in `self.header.hash` does not match the actual header's hash, since the header being constructed here is [populated from](https://github.com/paritytech/parity-ethereum/blob/master/ethcore/src/block.rs#L343) a subset of fields on the header. That works for verification since `self.header` only needs to match a [subset of fields](https://github.com/paritytech/parity-ethereum/blob/master/ethcore/src/verification/verification.rs#L250) on the imported header. But it does not work for the CSV if we want to have the actual header hash.